url8db16c7d62_Arsaell_EdT_tgz-pJ8-DATA_MainJ8/src/GUI/MainFrame.java

```diff
@@ -0,0 +1,588 @@
+import java.awt.event.FocusListener;
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.util.ArrayList;
+import java.util.Collections;
+
+import javax.imageio.ImageIO;
+import javax.swing.AbstractButton;
+import javax.swing.AbstractListModel;
+import javax.swing.ImageIcon;
+import javax.swing.JButton;
+import javax.swing.JFileChooser;
+import javax.swing.JFrame;
+import javax.swing.JList;
+import javax.swing.JOptionPane;
+import javax.swing.JPanel;
+import javax.swing.JScrollPane;
+import javax.swing.JSeparator;
+import javax.swing.JToolBar;
+import javax.swing.ListSelectionModel;
+import javax.swing.border.EmptyBorder;
+import javax.swing.event.ListSelectionEvent;
+import javax.swing.event.ListSelectionListener;
+import javax.swing.filechooser.FileNameExtensionFilter;
+
+import DATA.Classroom;
+import DATA.DataStore;
+import DATA.Group;
+import DATA.Teacher;
+
+public class MainFrame extends JFrame implements ActionListener {
+
+	private JPanel contentPane;
+	private DataStore dataStore;
+	private JButton btnSave;
+	private JButton btnOpen;
+	private JButton btnNew;
+	private JButton btnTeachers;
+	private AbstractButton btnClassrooms;
+	private JButton btnGroups;
+	private JButton btnFields;
+	private JButton btnWeekTable;
+	private JButton btnLancer;
+	private JButton btnFillFrame;
+	private JPanel listPanel;
+	private JPanel listPanel2;
+	private JPanel EdTPanel;
+	
+	public MainFrame(DataStore ds) {
+
+		setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+		setBounds(100, 100, 450, 300);
+		
+		this.dataStore = ds;
+		
+		contentPane = new ContentPane();
+		contentPane.setBorder(new EmptyBorder(5, 5, 5, 5));
+		contentPane.setLayout(new BorderLayout(0, 0));
+		setContentPane(contentPane);
+		setExtendedState(JFrame.MAXIMIZED_BOTH);
+		setVisible(true);
+		setResizable(true);
+		
+		JToolBar toolBar = new JToolBar();
+		toolBar.setFloatable(false);
+		toolBar.setMargin(new Insets(5, 5, 5, 5));
+		toolBar.setBackground(new Color(230, 230, 230));
+		getContentPane().add(toolBar, BorderLayout.NORTH);
+		
+		try {
+			btnSave = new JButton(new ImageIcon(ImageIO.read(new File("img/icons/disk.png"))));
+			btnSave.addActionListener(new ActionListener() {
+				public void actionPerformed(ActionEvent e) {
+					BufferedReader br;
+					try {
+						br = new BufferedReader(new FileReader(System.getProperty("user.dir") + "/dataStoreLocation.loc"));
+						String fileName = "";
+						String sCurrentLine = "";
+						while ((sCurrentLine = br.readLine()) != null) {
+							fileName = sCurrentLine;
+						}
+						File fichier =  new File(fileName) ;
+						 // ouverture d'un flux sur un fichier
+						ObjectOutputStream oos =  new ObjectOutputStream(new FileOutputStream(fichier)) ;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *oos*. Execution paths that do not contain closure statements were detected. Ensure *oos* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).